datatypes/src/raster/data_type.rs

```diff
@@ -88,6 +88,25 @@ pub enum RasterDataType {
     F64,
 }
 
+impl RasterDataType {
+    #[allow(clippy::float_cmp)]
+    #[allow(clippy::cast_lossless)]
+    pub fn is_valid(self, value: f64) -> bool {
+        match self {
+            RasterDataType::U8 => value as u8 as f64 == value,
+            RasterDataType::U16 => value as u16 as f64 == value,
+            RasterDataType::U32 => value as u32 as f64 == value,
+            RasterDataType::U64 => value as u64 as f64 == value,
+            RasterDataType::I8 => value as i8 as f64 == value,
+            RasterDataType::I16 => value as i16 as f64 == value,
+            RasterDataType::I32 => value as i32 as f64 == value,
+            RasterDataType::I64 => value as i64 as f64 == value,
+            RasterDataType::F32 => value as f32 as f64 == value,
+            RasterDataType::F64 => true,
+        }
+    }
+}
+
```

Comment:
Please add a comment on what this does since the back and forth-casting is not self-explanatory at all.

Reply:
3093c7cbd5f4681c875c20f1ecc64da2377494d5